nft/ethereum_payment_obligation.go

```diff
@@ -145,7 +263,21 @@ func (s *ethereumPaymentObligation) MintNFT(ctx context.Context, documentID []by
 		return nil, nil, err
 	}
 
-	registry := common.HexToAddress(registryAddress)
+	// if the role_proof is given check if the role exists with account id as collaborator
+	// at the moment, since we generate role key randomly, this will not work
+	// but once we figure out acls on the client API, this will continue working with out any chnages
+	if request.SubmitRoleProof != "" {
+		_, role := getRoleForAccount(cd, request.SubmitRoleProof, cid)
+		if role == nil {
+			return nil, nil, ErrNFTRoleMissing
+		}
+	}
+
+	if !request.GrantNFTReadAccess && request.SubmitNFTReadAccessProof {
+		return nil, nil, errors.New("enabled grant nft access to generate NFT read access proof")
```

Comment:
This error message doesn't really say what the problem is, I think it's not clear enough.

Reply:
Also, wouldn't it be better to do this rather basic check further up at the beginning before doing a bunch of other validation?

Reply:
This is the second validation check in the request. This is in the top-level validation

Reply:
Read the error message again. I don' think it makes sense. Shouldn't it be: `disabled GrantNFTReadAccess...`